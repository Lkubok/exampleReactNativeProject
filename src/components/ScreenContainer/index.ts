import Component from './ScreenContainer';

export default Component;
