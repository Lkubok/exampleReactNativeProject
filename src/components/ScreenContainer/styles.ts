import { StyleSheet, Dimensions } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';

import { variables } from 'res/variables';

const headerHeight = getStatusBarHeight();
const { width, height } = Dimensions.get('screen');

export default StyleSheet.create({
  container: {
    backgroundColor: variables.colors.white,
    height: height,
    paddingHorizontal: variables.spacing.large,
  },
  containerFullWidth: {
    backgroundColor: variables.colors.white,
    height,
    paddingHorizontal: 0,
  },
  headerHeadingMarginBottom: {
    marginBottom: variables.spacing.min,
  },
  headerMarginBottom: {
    marginBottom: variables.spacing.large,
  },
  headerMarginTop: {
    marginTop: variables.spacing.big,
  },
  navbar: {},
  navbarContainer: {
    backgroundColor: variables.colors.overlayHeaderBg,
    paddingTop: headerHeight,
    width,
    zIndex: 10,
  },
  safeAreaView: {},
});
