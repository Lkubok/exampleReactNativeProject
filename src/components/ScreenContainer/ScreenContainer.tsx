import React, { FunctionComponent, ReactNode, useMemo } from 'react';
import { StatusBar, StyleSheet, View } from 'react-native';

// const { height } = Dimensions.get('window');

import { Navbar } from 'components';
import { variables } from 'res/variables';
import { isIOS } from 'utils';

import styles from './styles';

interface Props {
  bgColor?: string;
  compact?: boolean;
  children: ReactNode;
  fullWidth?: boolean;
  navbarLeft?: ReactNode;
  navbarRight?: ReactNode;
  navbarCenter?: ReactNode;
  navbarBottom?: ReactNode;
  withBackButton?: boolean;
  withBottomNavigation?: boolean;
}

const ScreenContainer: FunctionComponent<Props> = ({
  bgColor,
  compact,
  children,
  fullWidth,
  navbarLeft,
  navbarRight,
  navbarCenter,
  navbarBottom,
  withBackButton,
  withBottomNavigation,
}) => {
  const barStyle = 'dark-content';

  const renderStatusBar = () =>
    isIOS ? (
      <StatusBar barStyle={barStyle} />
    ) : (
      <StatusBar backgroundColor={variables.colors.transparent} barStyle={barStyle} translucent={true} />
    );

  const renderNavbar = useMemo(
    () => (
      <View style={styles.navbarContainer}>
        <Navbar
          bottom={navbarBottom && navbarBottom}
          center={navbarCenter && navbarCenter}
          right={navbarRight}
          withLinkBack={withBackButton}
          left={navbarLeft}
        />
      </View>
    ),
    [navbarBottom, navbarCenter, navbarRight, navbarLeft, withBackButton],
  );

  // If there is bottom navigation - customize screen height
  let bottomNavigation = 0;
  if (withBottomNavigation) {
    if (isIOS) {
      bottomNavigation = variables.navbarBottomHeight.iOS;
    } else {
      bottomNavigation = variables.navbarBottomHeight.android;
    }
  }

  const navbar = navbarLeft || navbarRight || navbarCenter || withBackButton;

  // const safeAreaHeight = height - bottomNavigation;
  //TODO: consider restoring upper one line
  // const safeAreaHeight = Platform.OS === 'android' ? height : height - bottomNavigation - headerHeight;

  return (
    <View
      style={StyleSheet.flatten([
        styles.container,
        !!compact && { paddingHorizontal: variables.spacing.medium },
        !!fullWidth && styles.containerFullWidth,
        !!bgColor && { backgroundColor: bgColor },
        { paddingBottom: bottomNavigation },
      ])}>
      {renderStatusBar()}
      {navbar && renderNavbar}
      {/* <SafeAreaView style={StyleSheet.flatten([styles.safeAreaView, { height: safeAreaHeight }])}> */}
      {/* TODO: consider deleting code under and restore upper one */}
      {/* <SafeAreaView style={StyleSheet.flatten([styles.safeAreaView, { height: height }])}> */}
      {navbar && <View style={styles.navbar} />}
      {children}
      {/* </SafeAreaView> */}
    </View>
  );
};

export default ScreenContainer;
