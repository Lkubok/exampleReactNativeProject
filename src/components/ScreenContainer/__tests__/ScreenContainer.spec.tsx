import React from 'react';
import { act, customRender } from 'utils/testUtils';

import ScreenContainer from '../ScreenContainer';

jest.useFakeTimers();

describe('ScreenContainer component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: any;
  beforeEach(() => {
    props = {
      autoCapitalize: 'words',
      blocked: false,
      error: 'testError',
      initialFocus: true,
      isPasswordField: true,
      keyboardType: 'number-pad',
      label: 'test label',
      onBlur: () => {},
      onChange: () => {},
      onChangeText: () => {},
      onFocus: () => {},
      onSubmitEditing: () => {},
      placeholder: 'test placeholder',
      secureTextEntry: true,
      shakeCounter: 5,
      useNativeDriver: true,
      value: 'test value',
    };
  });

  it('should match snapshot when 1 line', () => {
    const snapshot = customRender(<ScreenContainer {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
