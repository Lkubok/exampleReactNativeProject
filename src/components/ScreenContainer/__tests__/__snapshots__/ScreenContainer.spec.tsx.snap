// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenContainer component should match snapshot when 1 line 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "[7m[38;2;255;255;255m#ffffff[39m[27m",
        "height": 1334,
        "paddingBottom": 0,
        "paddingHorizontal": 32,
      }
    }
  />
</RNCSafeAreaProvider>
`;
